### PR TITLE
Creating plotting.py with pandas-flavor

### PR DESCRIPTION
**Pull request by FloraSauerbronn**
decorating the plot transect method.

### CODE REVIEW

File: gliderpy/plotting.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    raise
+
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+from pandas_flavor import register_dataframe_method
+
+@register_dataframe_method
+
```
Comment (ocefpaf) [inline gliderpy/plotting.py:26]: Let's remove this blank link to make it easier to read that we are decorating this specific function. We will add more in the future and blank lines after a decorator can be confusing when there are many.
```suggestion
```

**Comment by ocefpaf** [comment]
@FloraSauerbronn do you mind adding a description above, you can edit your empty box, so we have something for future reference? You can go as simple as "decorating the plot transect method."

BTW, one place where you oceanography background will com in handy, is in implementing some improvements to this function. For example, you can add an if-clause to error out if the variable chosen for plotting is not in an expected list of vars that would work. At the moment I believe that we can only do `temperature` and `salinity`.

File: gliderpy/plotting.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    :param var: variable to colour the scatter plot
+    :return: figure, axes
+    """
+    cmap = kw.get("cmap", None)
+
+    fig, ax = plt.subplots(figsize=(17, 2))
+    cs = ax.scatter(
+        df.index,
+        df["depth (m)"],
```
[the remark this answers is not in the document]
Comment (FloraSauerbronn) [reply gliderpy/plotting.py:43]: I tested this locally, and it worked, but it was in a notebook (using what is in plotting.py)
`from gliderpy.fetchers import GliderDataFetcher

glider_grab = GliderDataFetcher()

glider_grab.fetcher.dataset_id = "whoi_406-20160902T1700"
df = glider_grab.to_pandas()
df.plot_transect(var="temperature")`

**Comment by FloraSauerbronn** [comment]
> @FloraSauerbronn do you mind adding a description above, you can edit your empty box, so we have something for future reference? You can go as simple as "decorating the plot transect method."
> 
> BTW, one place where you oceanography background will com in handy, is in implementing some improvements to this function. For example, you can add an if-clause to error out if the variable chosen for plotting is not in an expected list of vars that would work. At the moment I believe that we can only do `temperature` and `salinity`.

--> When you say 'not in an expected list of vars,' do you mean they have a lot of NaNs? Or just outliers? Because if it's just NaNs causing the problem, it's simple, but if it's outliers, then we would need to consider the values of the limit more carefully.

File: gliderpy/plotting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ocefpaf) [reply gliderpy/plotting.py:43]: I'm surprised b/c it shouldn't without the variable name change. Or did you change the variable name locally and did not commit it here?

**Comment by ocefpaf** [comment]
> About the if clause @ocefpaf
> 
> --> When you say 'not in an expected list of vars,' do you mean they have a lot of NaNs? Or just outliers? Because if it's just NaNs causing the problem, it's simple, but if it's outliers, then we would need to consider the values of the limit more carefully.

I mean the variables names. In the `plot_transect` we have one mandatory argument:

```
:param var: variable to colour the scatter plot
```

That can be either `salinity` or `temperature`, based on the list of [variables we have are fetching](https://github.com/ioos/gliderpy/blob/a71eaafdb975724155a8a8742580c9d9aba858d9/gliderpy/servers.py#L3-13):

```
server_vars = {
    "https://gliders.ioos.us/erddap": [
        "latitude",
        "longitude",
        "pressure",
        "profile_id",
        "salinity",
        "temperature",
        "time",
    ],
}
```

None of the others variables make sense to be in a section plot, right? However, let's keep this first PR simple and we can address that in a follow up one. reducing the context of the PRs will help us keep the momentum going and code changes/reviews easier to manage.

**Comment by ocefpaf** [comment]
pre-commit.ci autofix

**Pull request merged**